walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -31,22 +31,39 @@
       return spots;
     }
 
-    /** Deletes irrelevant events. */
-    eventsList.removeIf(event -> Collections.disjoint(request.getAttendees(), 
-                                                      event.getAttendees()));
+    eventsList.removeIf(event -> 
+        (Collections.disjoint(request.getAttendees(), event.getAttendees()) &&
+         Collections.disjoint(request.getOptionalAttendees(), event.getAttendees())));
 
-    if (eventsList.isEmpty() || request.getAttendees().isEmpty()) {
+    if (eventsList.isEmpty()) {
       return Arrays.asList(TimeRange.WHOLE_DAY);
     }
-    
+
     /** Sorts the events list by start time. */
     Collections.sort(eventsList, new Comparator<Event>(){
       public int compare(Event a, Event b) {
         return TimeRange.ORDER_BY_START.compare(a.getWhen(), b.getWhen());
       }
     });
     
+    if (request.getAttendees().isEmpty() || request.getOptionalAttendees().isEmpty()) {
+      return (Collection<TimeRange>)findMeetingSpots(eventsList, meetingDuration);
+    }
+
+    spots = findMeetingSpots(eventsList, meetingDuration);
+    if (spots.isEmpty()) {
+      /** When there are no meeting spots for both mandatory and optional attendees. */
+      eventsList.removeIf(event -> 
+          Collections.disjoint(request.getAttendees(), event.getAttendees()));
+      return (Collection<TimeRange>)findMeetingSpots(eventsList, meetingDuration);
```

Comment:
it seems that you don't need the conversion. The conversion could happen implicitly.